moveit_ros/planning/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -225,6 +218,10 @@ MoveItCpp::execute(const robot_trajectory::RobotTrajectoryPtr& robot_trajectory,
   robot_trajectory->getRobotTrajectoryMsg(robot_trajectory_msg);
   trajectory_execution_manager_->push(robot_trajectory_msg, controllers);
   trajectory_execution_manager_->execute();
+  if (!blocking)
```

Comment:
The reason why non-blocking does not make sense to me without the "non-blocking execution" is that you don't get the result of the execution and you cannot execute anything unless the execution is completed or canceled. 
So I am not sure about your use case. Could you explain a bit about it?